this was originally (now adapted for gentoo)
commit 23d5ce636b7897dc3f233a5293f3164dac50566d
Author: Jos van den Oever <jos@vandenoever.info>
Date:   Sun Jul 31 12:32:44 2011 +0200

    Revert patch that breaks parsing of PDF files.
    The offending patch was 672c8b7b3cb4400bc505421f4cc70cf742ed9df0.

diff -ruN strigi-0.7.6.orig/libstreamanalyzer/lib/pdf/pdfparser.cpp strigi-0.7.6/libstreamanalyzer/lib/pdf/pdfparser.cpp
--- strigi-0.7.6.orig/libstreamanalyzer/lib/pdf/pdfparser.cpp	2011-07-26 19:12:39.000000000 +0200
+++ strigi-0.7.6/libstreamanalyzer/lib/pdf/pdfparser.cpp	2011-10-19 00:18:51.583639819 +0200
@@ -35,12 +35,13 @@
 PdfParser::read(int32_t min, int32_t max) {
     int32_t off = (int32_t)(pos-start);
     int32_t d = (int32_t)(stream->position() - bufferStart);
-    bufferStart += off;
+    min += d;
+    if (max > 0) max += d;
     stream->reset(bufferStart);
     int32_t n = stream->read(start, min, max);
 //    printf("objstart %i %i\n", d, n);
     if (n < min) return stream->status();
-    pos = start;
+    pos = start + off;
     end = start + n;
     return Ok;
 }
